checks/process.go

```diff
@@ -106,28 +93,111 @@ func (p *ProcessCheck) Run(cfg *config.AgentConfig, groupID int32) ([]model.Mess
 	p.lastCPUTime = cpuTimes[0]
 	p.lastRun = time.Now()
 
-	statsd.Client.Gauge("datadog.process.containers.host_count", totalContainers, []string{}, 1)
-	statsd.Client.Gauge("datadog.process.processes.host_count", totalProcs, []string{}, 1)
+	statsd.Client.Gauge("datadog.process.containers.host_count", float64(totalContainers), []string{}, 1)
+	statsd.Client.Gauge("datadog.process.processes.host_count", float64(totalProcs), []string{}, 1)
 	log.Debugf("collected processes in %s", time.Now().Sub(start))
 	return messages, nil
 }
 
+func createProcCtrMessages(
+	procsByCtr map[string][]*model.Process,
+	containers []*model.Container,
+	cfg *config.AgentConfig,
+	sysInfo *model.SystemInfo,
+	groupID int32,
+) ([]model.MessageBody, int, int) {
+	totalProcs, totalContainers := 0, 0
+	msgs := make([]*model.CollectorProc, 0)
+
+	// we first split non-container processes in chunks
+	if procsByCtr[emptyCtrID] != nil {
+		totalProcs += len(procsByCtr[emptyCtrID])
+	}
+	chunks := chunkProcesses(procsByCtr[emptyCtrID], cfg.MaxPerMessage)
+	for _, c := range chunks {
+		msgs = append(msgs, &model.CollectorProc{
+			HostName:  cfg.HostName,
+			Info:      sysInfo,
+			Processes: c,
+			GroupId:   groupID,
+		})
+	}
+
+	// pack all containered processes with containers in a message
+	validIndex := 0
+	ctrProcs := make([]*model.Process, 0)
+	for _, ctr := range containers {
+		// if all processes are skipped for this container, don't send the container
+		if _, ok := procsByCtr[ctr.Id]; !ok {
```

Comment:
does this ever happen?

Reply:
If your container is running only one process and that process is blacklisted, would that be the case?

Reply:
Why are we doing this? We still want to send the container even if we're not collecting its processes, no?

Reply:
Would that cause confusion in the UI when people click a container and expecting processes to show up but never happens? I could change the logic as well and add something in UI to make it obvious.